src/file.cpp

```diff
@@ -336,14 +337,30 @@ namespace libtorrent
 		return int(size);
 	}
 
-#ifdef TORRENT_WINDOWS
-	std::string convert_separators(std::string p)
+	template <typename T>
+	std::string convert_from_native_path(T* s);
+
+	template <>
+	std::string convert_from_native_path(char const* s) { return convert_from_native(s); }
+
+	template <>
+	std::string convert_from_native_path(char* s) { return convert_from_native(s); }
```

Comment:
how does the overload taking a non-const char pointer differ from the one that take a regular const string? it doesn't actually do any in-place editing, does it?